RiverLoggerApi.Repository/DataContext.cs

```diff
@@ -0,0 +1,61 @@
+﻿using Dapper;
+using Microsoft.Extensions.Options;
+using MySql.Data.MySqlClient;
+using System.Data;
+
+namespace RiverLoggerApi.Repository
+{
+    public class DataContext
+    {
+        private DbSettings _dbSettings;
+
+        public DataContext(IOptions<DbSettings> dbSettings)
+        {
+            _dbSettings = dbSettings.Value;
+        }
+
+        public IDbConnection CreateConnection()
+        {
+            var connectionString = $"Server={_dbSettings.Server}; Database={_dbSettings.Database}; Uid={_dbSettings.UserId}; Pwd={_dbSettings.Password};";
+            return new MySqlConnection(connectionString);
+        }
+
+        public async Task Init()
+        {
+            await _initDatabase();
+            await _initTables();
+        }
+
+        private async Task _initDatabase()
+        {
+            // create database if it doesn't exist
+            var connectionString = $"Server={_dbSettings.Server};Port={_dbSettings.Port};Uid={_dbSettings.UserId};Pwd={_dbSettings.Password};";
+            using var connection = new MySqlConnection(connectionString);
+            var sql = $"CREATE DATABASE IF NOT EXISTS `{_dbSettings.Database}`;";
+            await connection.ExecuteAsync(sql);
+        }
+
+        private async Task _initTables()
+        {
+            // create tables if they don't exist
+            using var connection = CreateConnection();
+            await _initUsers();
+
+            async Task _initUsers()
```

Comment:
You have two email

Reply:
Don't know if you should rename like EmailConfirmation

Reply:
i will fix it good catch